Example/StripeConnectExample/StripeConnectExample/MainViewController.swift

```diff
@@ -102,24 +110,57 @@ class MainViewController: UITableViewController {
 
     /// Called when table row is selected
     func performAction(_ row: Row, cell: UITableViewCell) {
-        let viewControllerToPush: UIViewController
+        var viewControllerToPresent: UIViewController
 
+        // Create a view controller for the selected component
         switch row {
         case .onboarding:
             let savedOnboardingSettings = AppSettings.shared.onboardingSettings
-            viewControllerToPush = embeddedComponentManager.createAccountOnboardingViewController(
+            viewControllerToPresent = embeddedComponentManager.createAccountOnboardingViewController(
                 fullTermsOfServiceUrl: savedOnboardingSettings.fullTermsOfServiceUrl,
                 recipientTermsOfServiceUrl: savedOnboardingSettings.recipientTermsOfServiceUrl,
                 privacyPolicyUrl: savedOnboardingSettings.privacyPolicyUrl,
                 skipTermsOfServiceCollection: savedOnboardingSettings.skipTermsOfService.boolValue,
                 collectionOptions: savedOnboardingSettings.accountCollectionOptions)
         case .payouts:
-            viewControllerToPush = embeddedComponentManager.createPayoutsViewController()
+            viewControllerToPresent = embeddedComponentManager.createPayoutsViewController()
         }
 
-        viewControllerToPush.navigationItem.backButtonDisplayMode = .minimal
-        addChangeAppearanceButtonNavigationItem(to: viewControllerToPush)
-        navigationController?.pushViewController(viewControllerToPush, animated: true)
+        // Fetch ViewController presentation settings
+        let presentationSettings = AppSettings.shared.presentationSettings
+
+        if presentationSettings.embedInTabBar {
+            // Embed in a tab bar
+            let tabBarController = UITabBarController()
+            viewControllerToPresent.tabBarItem = .init(title: row.label, image: UIImage(systemName: "star"), tag: 0)
+
+            tabBarController.viewControllers = [viewControllerToPresent]
+
+            viewControllerToPresent = tabBarController
+        }
+
+        // Configure the component VC's navbar
+        viewControllerToPresent.navigationItem.backButtonDisplayMode = .minimal
+        addChangeAppearanceButtonNavigationItem(to: viewControllerToPresent)
+        viewControllerToPresent.title = row.label
+
+        if presentationSettings.presentationStyleIsPush {
+            // Push to navigation stack
+            navigationController?.pushViewController(viewControllerToPresent, animated: true)
+        } else {
+            // Modally present
+
+            // Add a close button
+            viewControllerToPresent.navigationItem.leftBarButtonItem = .init(systemItem: .close, primaryAction: .init(handler: { [weak viewControllerToPresent] _ in
+                viewControllerToPresent?.dismiss(animated: true)
+            }))
+
+            if presentationSettings.embedInNavBar {
```

Comment:
Should the tab bar be embedded in the nav bar?

Reply:
In practice, I think apps would always embed the navbar into the tabbar rather than what we're doing here, but technically both are valid and seem to result in the same visual appearance. 

I did it in this order because it made the logic easier to manage.